pkg/scheduler/objects/application.go

```diff
@@ -191,13 +190,15 @@ func NewApplication(siApp *si.AddApplicationRequest, ugi security.UserGroup, eve
 	app.rmID = rmID
 	app.appEvents = schedEvt.NewApplicationEvents(events.GetEventSystem())
 	app.appEvents.SendNewApplicationEvent(app.ApplicationID)
-	app.setNewMetrics()
 	return app
 }
 
-func (sa *Application) setNewMetrics() {
-	metrics.GetSchedulerMetrics().IncTotalApplicationsNew()
-	metrics.GetQueueMetrics(sa.GetQueuePath()).IncQueueApplicationsNew()
+func (sa *Application) SetNewMetrics() {
+	// ensure the queue is set before calling this function
+	if sa.GetQueuePath() != "" {
```

Comment:
Don't add a conditional, just execute `setNewMetrics()` inside `SetQueue()`. Callers should not need to be updated at all.

Reply:
Thanks @craigcondit , good suggestion, addressed in latest PR.